samples/nrf9160/mqtt_simple/src/main.c

```diff
@@ -388,23 +383,16 @@ static const uint8_t* client_id_get(void)
 		goto exit;
 	}
 
-#if defined(CONFIG_NRF_MODEM_LIB)
-	char imei_buf[CGSN_RESPONSE_LENGTH + 1];
-	int err;
+	char hw_id_buf[HW_ID_LEN] = {0};
 
-	err = nrf_modem_at_cmd(imei_buf, sizeof(imei_buf), "AT+CGSN");
-	if (err) {
-		LOG_ERR("Failed to obtain IMEI, error: %d", err);
+	if (!hw_id_get(hw_id_buf, ARRAY_SIZE(hw_id_buf))) {
```

Comment:
Could add a print here that prints the error code returned by `hw_id_get` in case it fails.

Reply:
done